lnwire/message_test.go

```diff
@@ -0,0 +1,844 @@
+	msgAll = append(msgAll, newMsgError(t, r))
+	msgAll = append(msgAll, newMsgPing(t, r))
+	msgAll = append(msgAll, newMsgPong(t, r))
+	msgAll = append(msgAll, newMsgOpenChannel(t, r))
+	msgAll = append(msgAll, newMsgAcceptChannel(t, r))
+	msgAll = append(msgAll, newMsgFundingCreated(t, r))
+	msgAll = append(msgAll, newMsgFundingSigned(t, r))
+	msgAll = append(msgAll, newMsgFundingLocked(t, r))
+	msgAll = append(msgAll, newMsgShutdown(t, r))
+	msgAll = append(msgAll, newMsgClosingSigned(t, r))
+	msgAll = append(msgAll, newMsgUpdateAddHTLC(t, r))
+	msgAll = append(msgAll, newMsgUpdateFulfillHTLC(t, r))
+	msgAll = append(msgAll, newMsgUpdateFailHTLC(t, r))
+	msgAll = append(msgAll, newMsgCommitSig(t, r))
+	msgAll = append(msgAll, newMsgRevokeAndAck(t, r))
+	msgAll = append(msgAll, newMsgUpdateFee(t, r))
+	msgAll = append(msgAll, newMsgUpdateFailMalformedHTLC(t, r))
+	msgAll = append(msgAll, newMsgChannelReestablish(t, r))
+	msgAll = append(msgAll, newMsgChannelAnnouncement(t, r))
+	msgAll = append(msgAll, newMsgNodeAnnouncement(t, r))
+	msgAll = append(msgAll, newMsgChannelUpdate(t, r))
+	msgAll = append(msgAll, newMsgAnnounceSignatures(t, r))
+	msgAll = append(msgAll, newMsgQueryShortChanIDs(t, r))
+	msgAll = append(msgAll, newMsgReplyShortChanIDsEnd(t, r))
+	msgAll = append(msgAll, newMsgQueryChannelRange(t, r))
+	msgAll = append(msgAll, newMsgReplyChannelRange(t, r))
+	msgAll = append(msgAll, newMsgGossipTimestampRange(t, r))
+	msgAll = append(msgAll, newMsgQueryShortChanIDsZlib(t, r))
+	msgAll = append(msgAll, newMsgReplyChannelRangeZlib(t, r))
+
+	return msgAll
+}
+
+func newMsgInit(t testing.TB) *lnwire.Init {
+	t.Helper()
+
+	return lnwire.NewInitMessage(
+		rawFeatureVector(),
+		rawFeatureVector(),
+	)
+}
+
+// newMsgOpenChannel creates a testing OpenChannel message.
+func newMsgOpenChannel(t testing.TB, r *rand.Rand) *lnwire.OpenChannel {
+	t.Helper()
+
+	msg := &lnwire.OpenChannel{
+		FundingAmount:        btcutil.Amount(r.Int63()),
+		PushAmount:           lnwire.MilliSatoshi(r.Int63()),
+		DustLimit:            btcutil.Amount(r.Int63()),
+		MaxValueInFlight:     lnwire.MilliSatoshi(r.Int63()),
+		ChannelReserve:       btcutil.Amount(r.Int63()),
+		HtlcMinimum:          lnwire.MilliSatoshi(r.Int63()),
+		FeePerKiloWeight:     uint32(r.Int31()),
+		CsvDelay:             uint16(r.Intn(1 << 16)),
+		MaxAcceptedHTLCs:     uint16(r.Intn(1 << 16)),
+		ChannelFlags:         lnwire.FundingFlag(uint8(r.Intn(1 << 8))),
+		FundingKey:           randPubKey(t),
+		RevocationPoint:      randPubKey(t),
+		PaymentPoint:         randPubKey(t),
+		DelayedPaymentPoint:  randPubKey(t),
+		HtlcPoint:            randPubKey(t),
+		FirstCommitmentPoint: randPubKey(t),
+		ExtraData:            createExtraData(t, r),
+	}
+
+	_, err := r.Read(msg.ChainHash[:])
+	require.NoError(t, err)
+
+	_, err = r.Read(msg.PendingChannelID[:])
+	require.NoError(t, err)
+
+	return msg
+}
+
+func newMsgAcceptChannel(t testing.TB, r *rand.Rand) *lnwire.AcceptChannel {
```

Comment:
Not a blocker, but we could possibly re-use the code we have to generate sample messages of each types for the `testing/quick` tests we use for serialization.

Reply:
Got it. I think I did copy that code and broke them into smaller functions. This could be a TODO in the future to let them share the same code.